File: Samples/iOS/Objective-C/ADGNativeSample/ADGNativeSample/Views/NativeAdView.m
//
//  NativeAdView.m
//  ADGNativeSample
//
//  Copyright © 2016年 supership. All rights reserved.
//

#import <ADG/ADGInformationIconView.h>
#import "NativeAdView.h"

@implementation NativeAdView

- (UIView *)createAdView:(ADGNativeAd *)nativeAd {
    // アイコン
    UIImageView *iconImageView = [[UIImageView alloc] initWithFrame:CGRectMake(4, 4, 30, 30)];
    if (nativeAd.iconImage.url.length > 0) {
        NSURL *iconImageUrl = [NSURL URLWithString:nativeAd.iconImage.url];
        NSData *iconImageData = [NSData dataWithContentsOfURL:iconImageUrl];
        iconImageView.image = [UIImage imageWithData:iconImageData];
    }

    // タイトル
    UILabel *titleLbl = [[UILabel alloc] initWithFrame:CGRectMake(38, 4, 258, 15)];
    if (nativeAd.title.text.length > 0) {
        titleLbl.numberOfLines = 1;
        titleLbl.font = [titleLbl.font fontWithSize:13];
        titleLbl.text = nativeAd.title.text;
    }

    // 広告マーク
    UILabel *adTextLabel = [[UILabel alloc] initWithFrame:CGRectMake(40, 20, 28, 14)];
    adTextLabel.textColor = [UIColor lightGrayColor];
    adTextLabel.font = [adTextLabel.font fontWithSize:11];
    adTextLabel.text = @"広告";

    // 本文
    UILabel *descriptionLbl = [[UILabel alloc] initWithFrame:CGRectMake(4, 30, 296, 40)];
    if (nativeAd.desc.value.length > 0) {
        descriptionLbl.numberOfLines = 2;
        descriptionLbl.font = [descriptionLbl.font fontWithSize:11];
        descriptionLbl.text = nativeAd.desc.value;
        descriptionLbl.textColor = [UIColor lightGrayColor];
    }

    // メインイメージ
    UIImageView *mainImageView = [[UIImageView alloc] initWithFrame:CGRectMake(4, 64, 292, 156)];
    if (nativeAd.mainImage.url.length > 0) {
        NSURL *mainUrl = [NSURL URLWithString:nativeAd.mainImage.url];
        NSData *mainImageData = [NSData dataWithContentsOfURL:mainUrl];
        mainImageView.image = [UIImage imageWithData:mainImageData];
        mainImageView.contentMode = UIViewContentModeScaleAspectFit;
        mainImageView.clipsToBounds = YES;
    }

    // インフォメーションアイコン
    ADGInformationIconView *infoIconView = [[ADGInformationIconView alloc] initWithNativeAd:nativeAd];
    if (infoIconView) {
        [mainImageView addSubview:infoIconView];
        [infoIconView updateFrameFromSuperview:UIRectCornerTopRight];
    }

    // スポンサー
    UILabel *sponsoredLbl = [[UILabel alloc] initWithFrame:CGRectMake(4, 228, 150, 20)];
    sponsoredLbl.numberOfLines = 1;
    sponsoredLbl.font = [sponsoredLbl.font fontWithSize:10];
    sponsoredLbl.textColor = [UIColor colorWithRed:0.7 green:0.7 blue:0.7 alpha:1.0];
    if (nativeAd.sponsored.value.length > 0) {
        sponsoredLbl.text = [NSString stringWithFormat:@"sponsored by %@", nativeAd.sponsored.value];
    } else {
        sponsoredLbl.text = @"sponsored";
    }

    // CTA
    UIButton *actionBtn = [[UIButton alloc] initWithFrame:CGRectMake(178, 223, 114, 25)];
    if (nativeAd.ctatext.value.length > 0) {
        [actionBtn setTitle:nativeAd.ctatext.value forState:UIControlStateNormal];
        [actionBtn setTitleColor:[UIColor colorWithRed:0.12 green:0.56 blue:1.00 alpha:1.0]
                        forState:UIControlStateNormal];
        [actionBtn.titleLabel setFont:[UIFont boldSystemFontOfSize:14]];
        actionBtn.backgroundColor = [UIColor whiteColor];
        actionBtn.layer.borderWidth = 1.0f;
        actionBtn.layer.borderColor = [[UIColor colorWithRed:0.12 green:0.56 blue:1.00 alpha:1.0] CGColor];
        actionBtn.layer.cornerRadius = 5.0f;
        actionBtn.titleEdgeInsets = UIEdgeInsetsMake(1.0f, 1.0f, 1.0f, 1.0f);
        actionBtn.titleLabel.adjustsFontSizeToFitWidth = YES;
        [nativeAd setTapEvent:actionBtn];  // ボタンへのタップ反応追加
    }

    UIView *nativeAdView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 300, 250)];
    [nativeAdView addSubview:iconImageView];
    [nativeAdView addSubview:titleLbl];
    [nativeAdView addSubview:adTextLabel];
    [nativeAdView addSubview:mainImageView];
    [nativeAdView addSubview:descriptionLbl];
    [nativeAdView addSubview:sponsoredLbl];
    [nativeAdView addSubview:actionBtn];

    return nativeAdView;
}

@end

File: Samples/iOS/Swift/ADGNativeSampleForSwift/ADGNativeSampleForSwift/Views/NativeAdView.swift
//
//  NativeAdView.swift
//  ADGNativeSampleForSwift
//
//  Created on 2016/06/10.
//  Copyright © 2016年 Supership. All rights reserved.
//

import UIKit

class NativeAdView: UIView {

    required init(coder aDecoder: NSCoder) {
        fatalError("not implemented")
    }

    init (adgManagerViewController: ADGManagerViewController, nativeAd: ADGNativeAd) {
        // 広告を貼り付けるViewを生成
        super.init(frame: CGRect(x: 0, y: 0, width: 300, height: 250))

        // アイコン
        if let urlString = nativeAd.iconImage?.url,
            let url = NSURL(string: urlString),
            let data = NSData(contentsOfURL: url) {
                let iconImageView = UIImageView(frame: CGRect(x: 4, y: 6, width: 30, height: 30))
                iconImageView.image = UIImage(data: data)
                self.addSubview(iconImageView)
        }

        // タイトル
        if let adTitle = nativeAd.title?.text where !adTitle.isEmpty {
            let titleLabel = UILabel(frame: CGRect(x: 38, y: 4, width: 258, height: 15))
            titleLabel.text = adTitle;
            titleLabel.numberOfLines = 1
            titleLabel.font = titleLabel.font.fontWithSize(13)
            self.addSubview(titleLabel)
        }

        // 広告マーク
        let adMarkLabel = UILabel(frame: CGRect(x: 38, y: 22, width: 28, height: 14))
        adMarkLabel.text = "広告"
        adMarkLabel.font = adMarkLabel.font.fontWithSize(11)
        adMarkLabel.textColor = UIColor.lightGrayColor()
        self.addSubview(adMarkLabel)

        // 本文
        if let adDesc = nativeAd.desc?.value where !adDesc.isEmpty {
            let descLabel = UILabel(frame: CGRect(x: 4, y: 30, width: 296, height: 40))
            descLabel.text = adDesc
            descLabel.numberOfLines = 2
            descLabel.font = descLabel.font.fontWithSize(11)
            descLabel.textColor = UIColor.lightGrayColor()
            self.addSubview(descLabel)
        }

        // 広告イメージ
        if let urlString = nativeAd.mainImage?.url,
            let url = NSURL(string: urlString),
            let data = NSData(contentsOfURL: url) {
                let imageView = UIImageView(frame: CGRect(x: 0, y: 65, width: 300, height: 156))
                imageView.image = UIImage(data: data)
                imageView.contentMode = UIViewContentMode.ScaleAspectFit
                imageView.clipsToBounds = true
                self.addSubview(imageView)
        }

        // スポンサー
        if let adSponsored = nativeAd.sponsored?.value {
            let sponsoredLabel = UILabel(frame: CGRect(x: 4, y: 226, width: 150, height: 20))
            if !adSponsored.isEmpty {
                sponsoredLabel.text = "sponsored by " + adSponsored
            } else {
                sponsoredLabel.text = "sponsored"
            }
            sponsoredLabel.numberOfLines = 1
            sponsoredLabel.font = sponsoredLabel.font.fontWithSize(10)
            sponsoredLabel.textColor = UIColor(red: 0.7, green: 0.7, blue: 0.7, alpha: 1.0)
            self.addSubview(sponsoredLabel)
        }

        // CTA
        if let adCTA = nativeAd.ctatext?.value where !adCTA.isEmpty {
            let actionButton = UIButton(frame: CGRect(x: 178, y: 223, width: 114, height: 25))
            actionButton.setTitle(adCTA, forState: UIControlState.Normal)
            actionButton.setTitleColor(UIColor(red: 0.12, green: 0.56, blue: 1.00, alpha: 1.0), forState: UIControlState.Normal)
            actionButton.titleLabel?.font = UIFont.boldSystemFontOfSize(14)
            actionButton.titleLabel?.adjustsFontSizeToFitWidth = true
            actionButton.titleEdgeInsets = UIEdgeInsets(top: 1.0, left: 1.0, bottom: 1.0, right: 1.0)
            actionButton.backgroundColor = UIColor.whiteColor()
            actionButton.layer.borderWidth = 1.0
            actionButton.layer.borderColor = UIColor(red: 0.12, green: 0.56, blue: 1.00, alpha: 1.0).CGColor
            actionButton.layer.cornerRadius = 5.0

            // ボタンへのタップ反応追加
            nativeAd.setTapEvent(actionButton)

            self.addSubview(actionButton)
        }

        // Viewへのタップ制御やローテーション制御
        // 必ず記述してください
        adgManagerViewController.delegateViewManagement(self, nativeAd: nativeAd)
    }
}
